src/main/java/org/akip/service/ProcessTimelineDefinitionService.java

```diff
@@ -0,0 +1,121 @@
+package org.akip.service;
+
+import org.akip.domain.ProcessDefinition;
+import org.akip.domain.ProcessTimelineDefinition;
+import org.akip.exception.BadRequestErrorException;
+import org.akip.repository.ProcessDefinitionRepository;
+import org.akip.repository.ProcessDeploymentRepository;
+import org.akip.repository.ProcessTimelineDefinitionRepository;
+import org.akip.service.dto.ProcessTimelineDefinitionDTO;
+import org.akip.service.dto.TaskInstanceDTO;
+import org.akip.service.mapper.ProcessTimelineDefinitionMapper;
+import org.camunda.bpm.model.bpmn.Bpmn;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.camunda.bpm.model.bpmn.instance.UserTask;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class ProcessTimelineDefinitionService {
+
+    @Autowired
+    private ProcessDefinitionRepository processDefinitionRepository;
+
+    @Autowired
+    private ProcessDeploymentRepository processDeploymentRepository;
+
+    @Autowired
+    private ProcessTimelineDefinitionRepository processTimelineDefinitionRepository;
+
+    private final ProcessTimelineDefinitionMapper processTimelineDefinitionMapper;
+
+    public ProcessTimelineDefinitionService(ProcessTimelineDefinitionMapper processTimelineDefinitionMapper) {
+        this.processTimelineDefinitionMapper = processTimelineDefinitionMapper;
+    }
+
+    public List<ProcessTimelineDefinitionDTO> getByBpmnProcessDefinitionId(String bpmnProcessDefinitionId) {
+        List<ProcessTimelineDefinitionDTO> processTimelineDefinitions = findAllByProcessDefinitionBpmnProcessDefinitionId(
```

Comment:
Remove the method `findAllByProcessDefinitionBpmnProcessDefinitionId` and call the repository here.